lldb/source/Plugins/ObjectFile/Minidump/ObjectFileMinidump.cpp

```diff
@@ -65,56 +66,53 @@ bool ObjectFileMinidump::SaveCore(const lldb::ProcessSP &process_sp,
   if (!process_sp)
     return false;
 
-  MinidumpFileBuilder builder;
+  llvm::Expected<lldb::FileUP> maybe_core_file = FileSystem::Instance().Open(
+      outfile, File::eOpenOptionWriteOnly | File::eOpenOptionCanCreate);
+  if (!maybe_core_file) {
+    error = maybe_core_file.takeError();
+    return false;
+  }
+  MinidumpFileBuilder builder(std::move(maybe_core_file.get()));
 
   Target &target = process_sp->GetTarget();
+  builder.AddHeaderAndCalculateDirectories(target, process_sp);
```

Comment:
If we construct the `builder` with a `process_sp`, it can use the `m_process_sp` instance variable that the `MinidumpFileBuilder` has to grab the target or process when it needs one.